File: 2016/19/solution.php
<?php

ini_set('memory_limit', '1G');

$elves = intval(fgets(STDIN));

$circle = new Ds\Deque(range(1, $elves));

while ($circle->count() > 1) {
    $circle->rotate(1);
    $circle->shift();
}

fwrite(STDOUT, $circle->first() . PHP_EOL);

$circle = new Ds\Deque(range(1, $elves));

$circle->rotate(intval($circle->count() / 2));

while ($circle->count() > 1) {
    $circle->shift();
    $circle->rotate($circle->count() % 2 === 1 ? 0 : 1);
}

fwrite(STDOUT, $circle->first() . PHP_EOL);

File: 2016/19/solution.py
import collections
import sys

elves = int(sys.stdin.readline().strip())

# Part 1
circle = collections.deque(range(1, elves + 1))

while len(circle) > 1:
    circle.rotate(-1)
    circle.popleft()

print(circle.popleft())

# Part 2
circle = collections.deque(range(1, elves + 1))
circle.rotate(-1 * (len(circle) // 2))

while len(circle) > 1:
    circle.popleft()
    circle.rotate((len(circle) % 2) - 1)

print(circle.popleft())
